app/src/main/java/at/bitfire/icsdroid/ui/EditCalendarActivity.kt

```diff
@@ -153,7 +153,7 @@ class EditCalendarActivity: AppCompatActivity() {
         model.active.value = calendar.isSynced
 
         val (username, password) = CalendarCredentials(this).get(calendar)
-        val requiresAuth = username != null && password != null
+        val requiresAuth = !username.isNullOrEmpty() && !password.isNullOrEmpty()
```

Comment:
Yes, exacty. Setting them to null will leave them empty, but not null. I guess that is because they have been created already.

Reply:
Furthermore I am unsure whether checking for Empty and not for Blank is the right approach, as this allows users to have username/password consisting of whitespace only. That is not very common though ...

Reply:
For clarity, the values should be null when they're not set. Maybe we can [trimToNull](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#trimToNull-java.lang.String-) the values when reading them from the UI?

Reply:
That is not possible, as `CredentialsModel.username` and `CredentialsModel.password` are both mutable live data, which cannot be null. However, I can try and make it work using the "requiresAuth" toggle, as discussed before, this would be more appropriate in any case, and would allow us to save whatever userinput has been made.